### ISSUE TITLE
[New Feature] Add total execution time in test report

### ISSUE THREAD

**Issue by sarathimca** [anshooarora/extentreports#13]
It would be great if we are adding total execution time in each test result grid and also displaying total execution time of entire script in test report.

Thanks
Partha

**Comment by anshooarora** [comment]
Partha,

This will be part of the java code, so no more js will be required.

You can see the latest sample here:  http://relevantcodes.com/Tools/ExtentReports2/Extent.html

* You can keep the ticket open until the release and close if it works as expected.

**Comment by anshooarora** [comment]
Partha,

Release 2.02 is here and it includes this feature.  Please close this ticket if everything looks good.

**Comment by sarathimca** [comment]
Anshoo,

Great, Perfect. Now each test execution time displays in corresponding grid. Actually I expected another one feature also with this ticket. Is there any possibility to display total test execution time in quick summary header ?.

**Comment by anshooarora** [comment]
Sure, I will make this a default feature starting next release.  In the interim, you can do this via javascript too.  Will attempt this tomorrow and share the script.

**Comment by anshooarora** [comment]
Partha,  here is the js and css you can use using <code>insertJs</code> and <code>insertCustomStyles</code>:

js:
```
var ended = $('.suite-ended-time').text().replace('-', '/').replace('-', '/');
var started = $('.suite-started-time').text().replace('-', '/').replace('-', '/');

var diff = new Date(new Date(ended) - new Date(started));
diff = diff.getMinutes() + 'm ' + diff.getSeconds() + 's';

$('.tests-quick-view .card-panel').append('<span class="total-time-taken">Total time taken: ' + diff + '</span>');
```

css:
```
.total-time-taken {
    float: right;
    margin-top: 30px;
}
```

**Comment by anshooarora** [comment]
Partha, here is the upcoming sample:

http://relevantcodes.com/Tools/ExtentReports2/Extent203.html

The total execution time is under the "Quick Test Summary".  Does this look good?

**Comment by sarathimca** [comment]
Yes Anshoo. it looks pretty good, Thanks a lot.

**Comment by vishwathma** [comment]
Bug
I executed 12 tests in total. These tests had just assert statements, no triggering of browser or something else. The total execution time displayed in the report(Quick Test Summary) is: Total time taken: 30m 1s. But the actual execution time was less than a minute for all 12 tests. Also, the time displayed for all the individual tests is 0h 0m 0s in the Details sec.
![wrong time](https://cloud.githubusercontent.com/assets/13534392/8925593/867aec7a-3526-11e5-98b2-7b7e44164a93.jpg)

![extent reports](https://cloud.githubusercontent.com/assets/13534392/8925570/4da43136-3526-11e5-8975-6f024bb1f9f9.png)

**Comment by anshooarora** [comment]
The bug for total-time-taken has been reported and I believe it could be because of zones.  Investigating this one.

The other issue I see for the time taken by each test case - are you sure your tests didn't take less than a second each to run?  As you said there are only asserts there.

**Comment by anshooarora** [comment]
Also, for the 2nd issue, are you using Extent with TestNG listener or as a logger?  If listener, the above is expected.. because the test is created and ended at the very same moment.

You will have to set the started and ended times for each test in which case:

```
ExtentTest test;
Date d;

SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy hh:mm:ss");

d = sdf.parse(getTestStartedTimeFromTestNGHere);
test.getTest().startedTime = d;

d = sdf.parse(getTestEndedTimeFromTestNGHere);
test.getTest().endedTime = d;
```

**Comment by anshooarora** [comment]
Can this issue be closed?  Its fixed in 2.031.

**Comment by anshooarora** [comment]
Issue fixed in 9345797. Closing ticket.
